tests/functional/multi_project/test_publication.py

```diff
@@ -0,0 +1,184 @@
+import pytest
+import pathlib
+
+from dbt.tests.util import run_dbt, get_artifact, write_file
+from dbt.contracts.publication import PublicationArtifact, PublicModel
+from dbt.exceptions import PublicationConfigNotFound, TargetNotFoundError
+
+
+model_one_sql = """
+select 1 as fun
+"""
+
+model_two_sql = """
+select fun from {{ ref('model_one') }}
+"""
+
+model_three_sql = """
+select fun from {{ ref('model_two') }}
+"""
+
+models_yml = """
+models:
+  - name: model_one
+    description: model one
+    access: public
+  - name: model_two
+    description: non-public model
+  - name: model_three
+    description: model three
+    access: public
+"""
+
+
+dependencies_yml = """
+projects:
+    - name: marketing
+"""
+
+marketing_pub_json = """
+{
+  "project_name": "marketing",
+  "metadata": {
+    "dbt_schema_version": "https://schemas.getdbt.com/dbt/publication/v1.json",
+    "dbt_version": "1.5.0",
+    "generated_at": "2023-04-13T17:17:58.128706Z",
+    "invocation_id": "56e3126f-78c7-470c-8eb0-c94af7c3eaac",
+    "env": {},
+    "adapter_type": "postgres",
+    "quoting": {
+      "database": true,
+      "schema": true,
+      "identifier": true
+    }
+  },
+  "public_models": {
+    "model.marketing.fct_one": {
+      "name": "fct_one",
+      "package_name": "marketing",
+      "unique_id": "model.marketing.fct_one",
+      "relation_name": '"dbt"."test_schema"."fct_one"',
+      "database": "dbt",
+      "schema": "test_schema",
+      "identifier": "fct_one",
+      "version": null,
+      "latest_version": null,
+      "public_node_dependencies": [],
+      "generated_at": "2023-04-13T17:17:58.128706Z",
+    },
+    "model.marketing.fct_two": {
+      "name": "fct_two",
+      "package_name": "marketing",
+      "unique_id": "model.marketing.fct_two",
+      "relation_name": '"dbt"."test_schema"."fct_two"',
+      "version": null,
+      "latest_version": null,
+      "public_node_dependencies": ["model.test.fct_one"],
+      "generated_at": "2023-04-13T17:17:58.128706Z",
+    }
+  },
+  "dependencies": []
+}
+"""
+
+ext_node_model_sql = """
+select * from {{ ref('marketing', 'fct_one') }}
+"""
+
+
+class TestPublicationArtifact:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "model_one.sql": model_one_sql,
+            "model_two.sql": model_two_sql,
+            "model_three.sql": model_three_sql,
+            "models.yml": models_yml,
+        }
+
+    def test_publication_artifact(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 3
+
+        publication_dict = get_artifact(project.project_root, "target", "test_publication.json")
+        publication = PublicationArtifact.from_dict(publication_dict)
+        assert publication
+        assert len(publication.public_models) == 2
+        assert publication.public_models["model.test.model_three"].public_node_dependencies == [
+            "model.test.model_one"
+        ]
```

Comment:
Could we add a test with a publication that contains dependencies as well?

Reply:
We need a bunch more tests. Haven't done a lot of them because in many cases it will take longer to write the test than the code, and I've been thinking about whether there's a way to include two projects in a test case, which our pytest fixtures makes pretty difficult.

Reply:
Have added check for dependencies in the PublicationArtifact and a test that actually uses two projects in the same test and copies the publication artifact.